src/constraint_solver/io.cc

```diff
@@ -180,6 +188,16 @@ class FirstPassVisitor : public ModelVisitor {
   }
 
  private:
+  //// TODO: TBD: not sure we are confusing terms with IntExpr, losing IntVar-iness, nevermind Boolean, etc...
```

Comment:
@lperron As for the delegation part, the only place the delegation function is being used is then. When ``IntVar`` is delegated as an ``IntExpr``. Not to discount the contributions that went into that, but maybe that's a couple of hours thinking it through? Maybe a few more following that through; i.e. rename the function or comment it out, and see what breaks. That would provide some indication as to the scope. Notwithstanding why do we need to delegate an ``IntVar:IntExpr`` to ``IntExpr`` at all?